app/routers/world_clock.py

```diff
@@ -0,0 +1,744 @@
+    """
+    if continent_name in get_timezones_parts(CONTINENT):
+        return continent_name
+
+
+def search_timezone_by_just_place(place_name: str) -> Optional[str]:
+    """Search for a timezone in the timezones list by a given place name.
+
+
+    Args:
+        place_name (str): The given place name.
+
+
+    Returns:
+        str: The suitable timezone.
+    """
+    res = [timezone for timezone in get_api_data(TIMEZONES_BASE_URL)
+           if place_name in timezone]
+    if len(res):
+        return res[UNPACK_ELEMENT]
+
+
+def generate_possible_timezone_path(map_hierarchy: Dict[str, Optional[str]]
+                                    ) -> List[Optional[str]]:
+    """Generate all possible timezone paths, given a map hierarchy
+       of continent -> country -> place.
+
+
+    Args:
+        map_hierarchy (dict): The hierarchy details.
+
+
+    Returns:
+        list: The list of possible timezones.
+    """
+    prefix = map_hierarchy[CONTINENT]
+    possibilities = [f"{prefix}/{map_hierarchy[COUNTRY]}",
+                     f"{prefix}/{map_hierarchy[PLACE]}"]
+    return possibilities
+
+
+def get_all_possible_timezone_paths_for_given_place(place_name: str)\
+        -> Optional[List[str]]:
+    """Get all possible timezone paths for given place.
+
+
+    Args:
+        place_name (str): The given place name.
+
+
+    Returns:
+        list: The list of possible timezones.
+    """
+    map_hierarchy = {CONTINENT: None, COUNTRY: None, PLACE: None, }
+    possibilities = []
+    country = get_country(place_name)
+    country = standardize_country_or_place(country)
+    if country:
+        map_hierarchy[COUNTRY] = country
+        map_hierarchy[PLACE] = place_name
+        continent = get_continent(country)
+        continent = standardize_continent(continent)
+        map_hierarchy[CONTINENT] = continent
+
+        if not continent:
+            if country in get_timezones_parts(PLACE):
+                possibilities.append(search_timezone_by_just_place(country))
+                return possibilities
+        else:
+            return possibilities + generate_possible_timezone_path(
+                map_hierarchy)
+    continent = get_continent(place_name)
+    continent = standardize_continent(continent)
+    if not continent:
+        place_name = standardize_country_or_place(place_name)
+        if place_name in get_timezones_parts('place'):
+            possibilities.append(search_timezone_by_just_place(place_name))
+            return possibilities
+    return possibilities.append(f"{continent}/{place_name}")
+
+
+def get_timezone_path_for_given_place(place_name: str) -> Optional[str]:
+    """Get a timezone path for a given place.
+
+
+    Args:
+        place_name (str): The given place name.
+
+
+    Returns:
+        str: The timezone path.
+    """
+    possibilities = \
+        get_all_possible_timezone_paths_for_given_place(place_name)
+    timezones_list = get_api_data(TIMEZONES_BASE_URL)
+    if possibilities:
```

Comment:
Returning always a list from `get_all_possible_timezone_paths_for_given_place` might save you the check here